### ISSUE TITLE
Keep records of time a client version last connected

### ISSUE THREAD

**Issue by edthamm** [factcast/factcast#1158]
**Is your feature request related to a problem? Please describe.**
When moving across versions of factcast there will be some point at which we will want to deprecate clients versions.

Lets say version 1.1 supports clients <=0.8 in version 1.2 client versions <= 0.9 are supported.
As an operator I would like to make sure that no one in my fleet still uses the now unsupported versions.

**Describe the solution you'd like**
Have a way to get the last time a client with a specific version connected. A simple thing like:

|Version| Last seen|
|---|---|
|0.8| 2020-09-12|

Would be enough

**Describe alternatives you've considered**

- Looking at logs or active sessions and extracting it from there. Hard to do/impractical imo.
- Scanning across all repositories in the company. Hard to do/impractical imo.

**Additional context**
None as of now.

**Comment by uweschaefer** [comment]
In order to know, whom to yell at, i'd gather as much information as possible from the client.

IP etc are a minimum, but i think a client id (think "SoupDelverySubsystem") will help a great deal.

**Comment by edthamm** [comment]
You know me the more the merrier although I usually only resort to yelling at long distances or at the late hours of the day :smiley: 

I assume that since we are talking about nails here those would and now please pardon my poor understanding we would create an event stream or do we do one per version so I can fetch the head of that?

**Comment by otbe** [comment]
In another ticket we implemented the possibility to send a `clientId` from the client to the server. Idea would be during the handshake procedure (if clientId is present) to emit a new metric a la `factcast_clients` with version and clientId as tags.

**Comment by uweschaefer** [comment]
@edthamm would a metric be a good solution to this?

**Comment by BernhardBln** [comment]
Merged to main, will be in master once #1660 is done

**Comment by uweschaefer** [comment]
Going for a different approach extending the handshake.
